libultraship/libultraship/Lib/Fast3D/gfx_sdl2.cpp

```diff
@@ -176,7 +176,8 @@ static void gfx_sdl_init(const char *game_name, bool start_in_fullscreen, uint32
 #ifndef __SWITCH__
     SDL_GL_GetDrawableSize(wnd, &window_width, &window_height);
 
-    if (start_in_fullscreen) {
+    bool steamDeckGameMode = (std::getenv("XDG_CURRENT_DESKTOP") != nullptr && std::string(std::getenv("XDG_CURRENT_DESKTOP")) == "gamescope");
+    if (start_in_fullscreen || steamDeckGameMode) {
```

Comment:
just curious, is there a specific reason to have `steamDeckGameMode` in here instead of using it to set `start_in_fullscreen`?

also, just want to ensure that checking for gamescope (which can be used outside of the steam deck/steam os) is reasonable for this (no use cases for windowed gamescope come to mind but i just want to make sure we're covering all the bases here)

Reply:
there are definitely people using gamescope outside of steam decks (and even [outside of steam](https://www.reddit.com/r/Lutris/comments/itj375/how_can_i_use_gamescope_in_lutris/))

i'm fine with using gamescope to detect steam decks, but if we do that we should probably rename `steamDeckGameMode` to `gamescopeFullscreen` or something, and also add documentation clarifying that running soh with gamescope will force fullscreen

Reply:
the only thing i've found for checking if a user is on the deck or not is via the steamworks api (which i'm pretty sure we're not going to want to include...)

Reply:
one idea would be to check if a user is on steamOS (i don't expect many steamOS users outside of the deck)